PurchasesCoreSwift/Purchasing/ProductInfoExtractor.swift

```diff
@@ -0,0 +1,114 @@
+//
+//  ProductInfoExtractor.swift
+//  PurchasesCoreSwift
+//
+//  Created by Juanpe Catalán on 8/7/21.
+//  Copyright © 2021 Purchases. All rights reserved.
+//
+
+import Foundation
+import StoreKit
+
+// TODO(post migration): Change back to internal and consider converting to a struct
+@objc(RCProductInfoExtractor)
+public class ProductInfoExtractor: NSObject {
+    @available(iOS 11.2, macOS 10.13.2, tvOS 11.2, *)
+    private lazy var isoPeriodFormatter = ISOPeriodFormatter()
+
+    // TODO(post migration): Change back to internal
+    @objc(extractInfoFromSKProduct:)
+    public func extractInfo(from product: SKProduct) -> ProductInfo? {
+        guard let currencyCode = product.priceLocale.rc_currencyCode() else {
+            return nil
```

Comment:
🤔 so for this one, to retain the same behavior as before, we'd want to set the currency code to `nil` if it can't be found. That way we're still sending everything else we know about the product to the backend. This is pretty similar to what we do for other optional properties

Reply:
Makes sense. In this case, I made the `currencyCode` optional, and then we're still sending everything else we know about the product to the backend.